opus/application/static_media/js/opus.js

```diff
@@ -724,12 +741,108 @@ var opus = {
+            // userAgent example output:
+            // Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/537.36
+            // (KHTML, like Gecko) Chrome/73.0.3683.103 Safari/537.36 OPR/60.0.3255.95
+            matchObj = userAgent.match(/OPR\/(\d+.\d+)/);
+            browserName = "Opera";
+            browserVersion = matchObj[1];
+        } else if (userAgent.indexOf("Chrome") > -1) {
+            // userAgent example output:
+            // Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/537.36
+            // (KHTML, like Gecko) Chrome/74.0.3729.169 Safari/537.36
+            matchObj = userAgent.match(/Chrome\/(\d+.\d+)/);
+            browserName = "Chrome";
+            browserVersion = matchObj[1];
+        } else if (userAgent.indexOf("Version") > -1) {
+            // userAgent example output:
+            // Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/605.1.15
+            // (KHTML, like Gecko) Version/12.1 Safari/605.1.15
+            matchObj = userAgent.match(/Version\/(\d+.\d+)/);
+            browserName = "Safari";
+            browserVersion = matchObj[1];
+        }
 
+        let modalMsg = (`Your current browser is ${browserName} ${browserVersion}.
+                        Please update the browser. OPUS supports Firefox (66+),
+                        Chrome (74+), Safari (12.1+), Opera (58+), and Microsoft Edge
+                        (18+).`);
+        $("#op-browser-version-msg .modal-body").html(modalMsg);
+        browserName = browserName.toLowerCase();
+
+        if (opus.browserSupport[browserName] === undefined) {
+            $("#op-browser-version-msg").modal("show");
+            return false;
+        } else {
+            if (parseFloat(browserVersion) < opus.browserSupport[browserName]) {
+                $("#op-browser-version-msg").modal("show");
+                return false;
+            }
+        }
+        return true;
+    },
+
+    checkBrowserSize: function() {
+        /**
+         * Check if browser width is less than 1280px or height is less
+         * than 275px. If so, display a modal to inform the user to
+         * resize the browser size.
+         */
+        if ($(window).width() < opus.browserSupport.width ||
+            $(window).height() < opus.browserSupport.height) {
+            $("#op-browser-size-msg").modal("show");
+        } else {
+            $("#op-browser-size-msg").modal("hide");
+        }
+    },
+
+    checkCookies: function() {
+        /**
+         * Check widgets cookie to determine if the user is a first time
+         * visitor. If so, we display a guide page.
+         * Note: we will call __help/splash.html api in the future to
+         * display the guide page. For now, we just show a modal.
+         */
+        if ($.cookie("visited") === undefined) {
+            // set the cookie for the first time user
+            $.cookie("visited", true, { expires: 28});  // days
```

Comment:
So the way this is currently implemented, a user can use OPUS every day  and then on the 28th day, get the splash dialog again, right? That seems wrong. We should either make the cookie never expire, or we should reset the 28 day counter every time the user runs OPUS (set it outside of the if).